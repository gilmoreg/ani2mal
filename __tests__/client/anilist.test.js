/* eslint-disable no-unused-vars */
/* globals describe, it, beforeEach, afterEach, expect */
const fetchMock = require('fetch-mock');
const Anilist = require('../../assets/anilist');
const { aniListResponse, aniListEmptyResponse } = require('../../helpers/fakes');

describe('Client side Anilist', () => {
  afterEach(() => {
    fetchMock.restore();
  });

  it('getList with one result', (done) => {
    fetchMock.mock('*', aniListResponse);
    Anilist.getList('test')
    .then((list) => {
      expect(list[0]).toBeDefined();
      done();
    });
  });

  it('getList with empty results', (done) => {
    fetchMock.mock('*', aniListEmptyResponse);
    Anilist.getList('test')
    .then((list) => {
      expect(list.length).toEqual(0);
      done();
    });
  });

  it('getList with network error', (done) => {
    fetchMock.mock('*', { throws: Error('timeout') });
    Anilist.getList('test')
    .then((list) => {
      expect(list.length).toEqual(0);
      done();
    });
  });
});
