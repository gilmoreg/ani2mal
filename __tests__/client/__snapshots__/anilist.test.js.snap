// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client side Anilist getList should return a valid list 1`] = `
Array [
  Object {
    "completedAt": Object {
      "day": null,
      "month": null,
      "year": null,
    },
    "id": 0,
    "progress": 12,
    "progressVolumes": undefined,
    "score": 10,
    "startedAt": Object {
      "day": 1,
      "month": 1,
      "year": 2017,
    },
    "status": "COMPLETED",
    "title": "test",
    "type": "anime",
  },
  Object {
    "completedAt": Object {
      "day": null,
      "month": null,
      "year": null,
    },
    "id": 1,
    "progress": 0,
    "progressVolumes": undefined,
    "score": 0,
    "startedAt": Object {
      "day": null,
      "month": null,
      "year": null,
    },
    "status": "CURRENT",
    "title": "test",
    "type": "anime",
  },
  Object {
    "completedAt": Object {
      "day": null,
      "month": null,
      "year": null,
    },
    "id": 0,
    "progress": 12,
    "progressVolumes": 1,
    "score": 10,
    "startedAt": Object {
      "day": 1,
      "month": 1,
      "year": 2017,
    },
    "status": "COMPLETED",
    "title": "test",
    "type": "manga",
  },
  Object {
    "completedAt": Object {
      "day": null,
      "month": null,
      "year": null,
    },
    "id": 1,
    "progress": 0,
    "progressVolumes": 0,
    "score": 0,
    "startedAt": Object {
      "day": null,
      "month": null,
      "year": null,
    },
    "status": "CURRENT",
    "title": "test",
    "type": "manga",
  },
]
`;

exports[`Client side Anilist getList with empty results 1`] = `"No data found for user test"`;

exports[`Client side Anilist getList with network error 1`] = `"No data found for user test"`;
